### PR TITLE
Underslung Grenade Launcher, Changes to Secondary_Weapons, Vendors

### PR DESCRIPTION
**Pull request by 5crownik007**
🆑 5crownik007
rscadd: 40mm Grenade launcher is now available for use. It can be found in Attachment Vendors. 40mm comes in 5 varieties, Practice, High Explosive, Frag, Smoke and Illumination. Illumination shells are currently nonfunctional.
bugfix: Covenant Smokebomb is now in its vendor
bugfix: Changes to secondary weapon code mean they can be unloaded manually and now eject spent casings 
bugfix: Rubber M7 SMG Magazine now uses appropriate sprites. M7 bullets now do 20 shield damage instead of 10
/🆑

### CODE REVIEW

**Review by caelaislinn — APPROVED** [review]

**Comment by 5crownik007** [comment]
Seems about right to me

File: code/modules/halo/weapons/ammo.dm [hunk 1]
```diff
@@ -445,3 +445,95 @@
...
+		playsound(src.loc, 'sound/effects/explosion1.ogg', 30, 1, -3)
+		src.fragmentate(get_turf(target), 50, 7, list(/obj/item/projectile/bullet/pellet/fragment = 1))
+	..()
+
+/obj/item/projectile/bullet/g40mm/smoke
+	damage = 30
+	armor_penetration = 5
+
+/obj/item/weapon/grenade/smokebomb/New()
+	..()
```
Comment (X0-11) [inline code/modules/halo/weapons/ammo.dm]: What's this here for?

Comment (5crownik007) [reply code/modules/halo/weapons/ammo.dm]: Appears to be legacy code from me experimenting with smokebombs
I'll remove it real quick

**Comment by 5crownik007** [comment]
All checks passed, seems ready to merge

**Comment by bloxgate** [comment]
Changelog needs some changes, info sent on Discord

**Comment by 5crownik007** [comment]
> 
> 
> Changelog needs some changes, info sent on Discord

Whoops I went to sleep immediately after I made my last changes

Should be good now

**Pull request merged**